TicTacToe/main.cpp

```diff
@@ -0,0 +1,79 @@
+#include <iostream>
+#include <queue>
+#include <string>
+#include <cassert>
+#include "PlayField.h"
+#include "TreeNode.h"
+
+struct Counter {
+    int crossesWin = 0;
+    int noughtsWin = 0;
+    int draw = 0;
+
+    Counter operator+(Counter elem) {
+        Counter res = *this;
+        res.crossesWin += elem.crossesWin;
+        res.noughtsWin += elem.noughtsWin;
+        res.draw += elem.draw;
+        return res;
+    }
+
+     friend std::ostream& operator<<(std::ostream &out, Counter &c)  {
+         out << "X " << c.crossesWin << ", 0 " << c.noughtsWin << ", D " << c.draw << std::endl;
+         return out;
+    }
+
+    int overall(){
+        return crossesWin + noughtsWin + draw;
+    }
+};
+
+void BuildSubTree(TreeNode& node) {
+    if(node.value().checkFieldStatus() == PlayField::fsNormal)
+        for(auto item : node.value().getEmptyCells()) {
+            node.addChild(node.value().makeMove(item));
+            BuildSubTree(node[node.childCount()-1]);
+        }
+}
+
+void WalkTree(const TreeNode& node) {
+    auto totalCount = Counter();
+    for(int i = 0; i < node.childCount(); ++i) {
+        auto localCount = Counter();
+        std::queue<TreeNode> queue;
```

Comment:
```suggestion
        std::queue<*TreeNode> queue;
```
Копии брать неразумно. С деструкторами намучаетесь, всё будет падать. У вас там указатели на объекты внутри, объекты надо потом удалять... И надо либо определять конструктор копирования, который будет не просто копировать указатели, но делать под них новые объекты. А сейчас указатели просто копируются. А деструктор, который чистит указатели, делать всёравно надо.
Тут лучше либо указатели, либо ссылки (не уверен, но думаю можно)...